Calculate base store taxes including the store's zip and city

james-allan opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->
This PR ensures that calculating order totals includes the store's postcode and city when taxes are calculated from the store's base address.

This is already the case when calculating taxes in other places, including [`WC_Tax::get_tax_location()`](https://github.com/woocommerce/woocommerce/blob/3.6.2/includes/class-wc-tax.php#L470-L471), [`WC_Customer::get_taxable_address()`](https://github.com/woocommerce/woocommerce/blob/3.6.2/includes/class-wc-customer.php#L194-L198), and [`WC_Tax::get_base_tax_rates()`](https://github.com/woocommerce/woocommerce/blob/3.6.2/includes/class-wc-tax.php#L517-L521)

All of these other cases use all the individual `WC()->countries` getters rather than `wc_get_base_location()` which `WC_order::get_tax_location()` uses. I wasn't sure if I should replace the `wc_get_base_location()` here or not. `wc_get_base_location()` is filterable so I figured I'd leave it but let me know if this should be consistant. 

### How to test the changes in this Pull Request:

1. Under **WooCommerce > Settings > Taxes > Tax Options** set _Calculate tax based on - Shop base address_: https://cl.ly/3184f24ff884
1. Set your shop base address: https://cl.ly/af760142cb92
1. Create a **Standard tax rate** that matches your shop base address criteria: https://cl.ly/4306f1a202cf. **Be sure to include a postcode and city.** 
1. Go to **WooCommerce > Orders > Add Order**
1. Create an order
1. Add a taxable product
1. Click **recalculate**:
   - on `master` or `3.6.2` no tax is added.
   - on this branch, taxes are added.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

Include the store's base postcode and city when calculating order taxes.

james-allan committed:
Calculate base store taxes including the store's zip and city

claudiosanches commented:
> All of these other cases use all the individual WC()->countries getters rather than wc_get_base_location() which WC_order::get_tax_location() uses. I wasn't sure if I should replace the wc_get_base_location() here or not. wc_get_base_location() is filterable so I figured I'd leave it but let me know if this should be consistant.

`wc_get_base_location()` it's filterable, but it's used inside `WC_Countries`, see:

https://github.com/woocommerce/woocommerce/blob/f0dcb5f67feef904cff50b983373cb6ad49e94de/includes/class-wc-countries.php#L190-L208

So no problem in replacing, since the filter should still works.

james-allan committed:
Replace wc_get_base_location() with individual getters

james-allan commented:
> So no problem in replacing, since the filter should still works.

Ah nice. In that case, I've removed the `wc_get_base_location()` and replaced it with the individual getters.

claudiosanches reviewed (approved):
Thanks for your help!
